### PR TITLE
Pull request to sync up the working system to upstream project

### PR DESCRIPTION
**Pull request by jdow**

### CODE REVIEW

**Comment by tinnightcap** [comment]
I am curious why you have added the helper scripts directly instead of including nubis-wrapper as a submodule? If you adapt the submodule method than I have no further questions. If you wish to keep this layout than I will have a few more :) (specifically issues that were already addressed in that repository).

File: README.md [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    ```
+
+* Get ec2 instance private IP
+    ```bash
+    $ aws ec2 describe-instances --filter "Name=tag:ServiceName,Values=nubis-elk" | jq ".Reservations|.[].Instances|.[].PrivateIpAddress" -r
+    ```
+
+* Get stack build information
+    ```bash
+    STACK_NAME="nubis-elk"; watch -n 1 "echo 'Container Stack'; aws cloudformation describe-stacks --query 'Stacks[*].[StackName,StackStatus]' --output text --stack-name $STACK_NAME; echo \"\nNested Stacks\"; aws cloudformation describe-stack-resources --stack-name $STACK_NAME --query 'StackResources[*].[LogicalResourceId, ResourceStatus]' --output text"
```
Comment (tinnightcap) [inline README.md]: FYI, if you are using the nubis-wrapper repository as a submodule there is a status.sh script now that takes care of this.

File: bin/.gitignore [hunk 2]
```diff
@@ -0,0 +1 @@
+variables.sh
```
Comment (tinnightcap) [inline bin/.gitignore]: While this works it is most common to place all ignores into a top level .gitignore file. AKA in that file you would list this as 'bin/variables.sh'. This makes it more clear to anyone looking at your project where all the secrets and such are located.

File: nubis/cloudformation/README.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+## Commands to work with CloudFormation
+NOTE: All examples run from the top level project directory.
+
+In these examples the stack is called *nubis-skel*. You will need to choose a unique name for your stack as their can only be one *nubis-skel* stack at a time.
+
+### Create
+To create a new stack:
+```bash
+aws cloudformation create-stack --template-body file://nubis/cloudformation/main.json --parameters file://nubis/cloudformation/parameters.json --stack-name nubis-skel
```
Comment (tinnightcap) [inline nubis/cloudformation/README.md]: You should either change these commands so they work with your project (change nubis-skel to your project/stack example name) or change these completely to use your new method with the wrapper scripts.

File: nubis/cloudformation/main.json [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+        "sandbox"
+      ],
+      "ConstraintDescription": "Must specify prod, stage, or sandbox."
+    },
+    "BaseZone": {
+      "Description": "Base zone to create route53 zone in",
+      "Default": "nubis.allizom.org",
+      "Type": "String"
+    },
+    "KeyName": {
```
Comment (tinnightcap) [inline nubis/cloudformation/main.json]: We changed all of these to be 'SSHKeyName', please change this to match the new convention.

File: nubis/cloudformation/main.json [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+            "Ref": "TechnicalOwner"
+          },
+          "Environment": {
+            "Ref": "Environment"
+          },
+          "AmiId": {
+            "Ref": "AmiId"
+          },
+          "KeyName": {
+            "Ref": "KeyName"
```
Comment (tinnightcap) [inline nubis/cloudformation/main.json]: This would then change to:
"Ref": "SSHKeyName"

File: nubis/cloudformation/parameters.json-dist [hunk 6]
```diff
@@ -0,0 +1,22 @@
+[
+  {
+    "ParameterKey": "ServiceName",
+    "ParameterValue": "skel"
```
Comment (tinnightcap) [inline nubis/cloudformation/parameters.json-dist]: This should be set to the service name for this project.

File: nubis/cloudformation/parameters.json-dist [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+  }, 
+  {
+    "ParameterKey": "Environment",
+    "ParameterValue": "sandbox"
+  }, 
+  {
+    "ParameterKey": "KeyName",
```
Comment (tinnightcap) [inline nubis/cloudformation/parameters.json-dist]: This should change to:
"Ref": "SSHKeyName"

File: nubis/builder/provisioner.json [hunk 8]
```diff
@@ -0,0 +1,19 @@
+{
```
Comment (gozer) [inline nubis/builder/provisioner.json]: This pattern can now be greatly simplified with the built-in support for [nubis/puppet/files](https://github.com/Nubisproject/nubis-docs/blob/master/PUPPET.md#nubispuppetfiles)

Just drop your migrate script in there, and your puppet manifests can access it as:

    "puppet:///nubis/files/migrate"

Meaning in this case, you can get rid of your custom provisionner entirely, and just install your migrate script via puppet.

**Comment by jdow** [comment]
I've taken the comments into account and will be submitting a new pull request shortly.

**Pull request closed without merging**